### ISSUE TITLE
Make render/pick tests use the scene

### ISSUE THREAD

**Issue by pjcozzi** [AnalyticalGraphicsInc/cesium#1259]
Most old low-level primitive rendering tests that use [`render`](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Specs/render.js) directly can be updated to use the real Cesium `Scene`, like [`DebugModelMatrixPrimitiveSpec`](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Specs/Scene/DebugModelMatrixPrimitiveSpec.js) does, which makes the tests more realistic and more concise.

**Comment by pjcozzi** [comment]
Billboards/Labels were updated in #2669

**Comment by pjcozzi** [comment]
@lilleyse after #817, we'll start chipping away at this (not all at once, we'll split it into several pull requests).  To see the files that need to be updated, search in all files for `Specs/render`.  An easy file to start with is _probably_ ViewportQuadSpec.js.  Look at DebugModelMatrixPrimitiveSpec.js for a simple example.

**Comment by pjcozzi** [comment]
@lilleyse let's start with `ViewportQuadSpec` and `GroundPrimitiveSpec` today.

* [ ] DebugAppearanceSpec
* [ ] EllipsoidPrimitiveSpec
* [ ] EllipsoidSurfaceAppearanceSpec
* [ ] GeometryRenderingSpec
* [ ] GlobeSpec
* [ ] GlobeSurfaceTileProviderSpec
* [ ] GroundPrimitiveSpec `// Just remove the reference; it is not used`
* [ ] MaterialAppearanceSpec
* [ ] MaterialSpec
* [ ] PerInstanceColorAppearanceSpec
* [ ] PolygonSpec
* [ ] PolylineCollectionSpec
* [ ] PolylineColorAppearanceSpec
* [ ] PolylineMaterialAppearanceSpec
* [ ] PrimitiveCollectionSpec
* [ ] PrimitiveCullingSpec
* [ ] PrimitiveSpec
* [ ] RectanglePrimitiveSpec
* [ ] SceneSpec `// Not sure we'll do this one`
* [ ] ViewportQuadSpec

**Comment by lilleyse** [comment]
Should all these pull requests go into their own branch, like renderer-refactor in #640 ?

**Comment by pjcozzi** [comment]
Yes, one branch per pull request.  The branch may have one or more refactored specs.

**Comment by lilleyse** [comment]
I mean should the master repo have a branch called spec-refactor, like how I made my renderer pull requests go into renderer-refactor?

**Comment by pjcozzi** [comment]
Ah, sorry.  No, it is fine to open the pull request directly into master.

**Comment by lilleyse** [comment]
GlobeSpec was updated in 8f20940.

**Comment by pjcozzi** [comment]
@lilleyse can you also update the new tests in PrimitiveSpec.js in the `3d-tiles` branch?
```
[runJsHint] Specs/Scene/PrimitiveSpec.js: line 402, col 9, 'frameState' is not defined.
[runJsHint] Specs/Scene/PrimitiveSpec.js: line 403, col 9, 'frameState' is not defined.
[runJsHint] Specs/Scene/PrimitiveSpec.js: line 404, col 9, 'us' is not defined.
[runJsHint] Specs/Scene/PrimitiveSpec.js: line 404, col 28, 'frameState' is not defined.
[runJsHint] Specs/Scene/PrimitiveSpec.js: line 406, col 9, 'ClearCommand' is not defined.
[runJsHint] Specs/Scene/PrimitiveSpec.js: line 409, col 9, 'render' is not defined.
[runJsHint] Specs/Scene/PrimitiveSpec.js: line 409, col 25, 'frameState' is not defined.
[runJsHint] Specs/Scene/PrimitiveSpec.js: line 439, col 39, 'frameState' is not defined.
```

**Comment by pjcozzi** [comment]
@lilleyse can we close this?  If not, what is left?  I took a quick look at SceneSpec and I'm not sure that we need to make any changes there.  Do we still need to update tests to use `renderForSpecs`?

**Comment by lilleyse** [comment]
@lasalvavida is working on a final cleanup PR for any files not included on this list, so until that is ready we can leave this open.

**Comment by lilleyse** [comment]
Thanks for the great work on this @lasalvavida.
